### PR TITLE
GH-106057: Handle recursion errors in inline class calls properly.

### PR DESCRIPTION
**Pull request by markshannon**
An obscure and new bug, so I'm skipping news.

* Issue: gh-106057

### CODE REVIEW

File: Lib/test/test_class.py [hunk 1]
```diff
@@ -740,6 +740,24 @@ class A(0, 1, 2, 3, 4, 5, 6, 7, **d): pass
         class A(0, *range(1, 8), **d, foo='bar'): pass
         self.assertEqual(A, (tuple(range(8)), {'foo': 'bar'}))
 
+    def testClassCallRecursionLimit(self):
```
Comment (Fidget-Spinner) [inline Lib/test/test_class.py:743]: Test fails on Windows 32 bit. Possibly because MSVC's stack usage on unoptimized builds is enormous.

Comment (markshannon) [reply Lib/test/test_class.py:743]: We aren't consuming C stack, so that shouldn't matter.

File: Lib/test/test_class.py [hunk 2]
```diff
@@ -740,6 +740,25 @@ class A(0, 1, 2, 3, 4, 5, 6, 7, **d): pass
...
+        class C:
+            def __init__(self):
+                self.c = C()
+
+        try:
+            C()
+        except RecursionError:
+            # OK
+            pass
```
Comment (brandtbucher) [inline Lib/test/test_class.py]:
```suggestion
        with self.assertRaises(RecursionError):
            C()
```

File: Lib/test/test_class.py [hunk 3]
```diff
@@ -740,6 +740,25 @@ class A(0, 1, 2, 3, 4, 5, 6, 7, **d): pass
...
+        def add_one_level():
+            #Each call to C() consumes 2 levels, so offset by 1.
+            C()
+        try:
+            add_one_level()
+        except RecursionError:
+            # OK
+            pass
```
Comment (brandtbucher) [inline Lib/test/test_class.py]:
```suggestion
        with self.assertRaises(RecursionError):
            add_one_level()
```

**Review by brandtbucher — COMMENTED** [review]
Looks good, just one suggestion for the test:

**Pull request merged**